### PR TITLE
Introducing ble adaptation for webOS

### PR DESCRIPTION
**Pull request by sungho1shin**
#### Problem
What is being fixed?  Examples:
* There is no support for ble-wifi commissioning on webOS.

#### Change overview
* Implement base function regarding ble behavior using webOS ble API
* Implement ble central functionality for webOS

#### Testing
How was this tested? (at least one bullet point required)
* If unit tests were added, how do they cover this issue?
* If unit tests existed, how were they fixed/modified to prevent this in future?
* If new unit tests are not added, why not?
* If integration tests were added, how do they verify this change?
* If new integration tests are not added, why not?
* If manually tested, what platforms controller and device platforms were manually tested, and how?
Controller: webOS OSE on rapsberrypi4
Controlee: ubuntu on rapsberrypi4 where connectedhomeip is compiled
It has been verified that ble-wifi commissioning works as expected between webOS (raspberrypi4) and ubuntu (rapsberrypi4)
* If no testing is required, why not?

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]
Approving since most of the delta is platform code.

I did not review the platform code though - this is 12K of new lines.

File: src/platform/webos/PlatformManagerImpl.cpp [hunk 1]
```diff
@@ -0,0 +1,581 @@
...
+        PlatformMgrImpl().HandleSwitchEvent(Switch::Events::SwitchLatched::Id);
+        break;
+    default:
+        break;
+    }
+}
+
+#if CHIP_WITH_GIO
+
+// LSHandle *Handle = NULL;
```
Comment (andy31415) [inline src/platform/webos/PlatformManagerImpl.cpp:90]: can we delete commented-out code?

File: src/platform/webos/PlatformManagerImpl.cpp [hunk 2]
```diff
@@ -0,0 +1,581 @@
...
+    labelList.add(floor);
+    labelList.add(direction);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR
+PlatformManagerImpl::_GetSupportedLocales(AttributeList<chip::CharSpan, kMaxLanguageTags> & supportedLocales)
+{
+    // In Linux simulation, return following hardcoded list of Strings that are valid values for the ActiveLocale.
```
Comment (andy31415) [inline src/platform/webos/PlatformManagerImpl.cpp:371]: this is webos, should it reference linux?

File: src/platform/webos/PlatformManagerImpl.cpp (hunk 1 above)
Comment (sungho1shin) [reply src/platform/webos/PlatformManagerImpl.cpp]: I've deleted the commented-out code and pushed the source code at https://github.com/project-chip/connectedhomeip/pull/17305/commits/49f0524441a16c76ad611ce0a26ac07d664e3e2a.

File: src/platform/webos/PlatformManagerImpl.cpp (hunk 2 above)
Comment (sungho1shin) [reply src/platform/webos/PlatformManagerImpl.cpp:304]: Yes, it referenced linux side and we verified it on webOS platform.

**Comment by sungho1shin** [comment]
Dear @andy31415,
First of all, thank you for your comments.

As you know, I've left the note for the comments.
This commit was for webOS platform and it was verified on webOS platform as expected.

**Comment by cabin15** [comment]
As I'm not able to trigger the CI build, it would be so kind if anyone approve running the CI for this PR. :)

**Comment by cabin15** [comment]
@andy31415 
Could you do us a favor trigger CI build?
I always appreciate your help.

**Pull request closed without merging**